### ISSUE TITLE
VariantRecordSample Documentation

### ISSUE THREAD

**Issue by mrvollger** [pysam-developers/pysam#407]
I cannot find any documentation for the class VariantRecordSample. Is there anywhere I can find a list of its attributes and functions. I know this is probably a silly question but I have spent quite some time looking and I cannot find it.

Thanks in advance!

**Comment by bioinformed** [comment]
Thanks for the report!

The first and most obvious problem is that the Sphinx docs don't include the `VariantRecordSample` object, so I'll definitely look into that.

In the mean time: The `VariantRecordSample` object implements a mapping-like object for a specific VCF/BCF sample row and column.  The keys are FORMAT fields and the values are the values in the VCF/BCF file.  There is special handling for "GT", through the `allele`, `allele_indices` and `phased` attributes.  There is also a `name` property that provides the sample's name.  Obviously, there are dozens of useful attributes and properties that could be added that would be of general utility, some part of htslib and others that are logical extensions.  I just haven't gotten to doing that yet.

**Comment by mvollger** [comment]
Thank you for the very speedy responce, this will be quite useful for me. 

Just one note, I think based it is not myVariantRecordSample.allele, but VariantRecordSample.alleles with an "s". At least the first one does not work for me and the second one does. 

Thanks again!

**Comment by fiapinto** [comment]
Hi!
Is there already a documentation for the class VariantRecordSample?
The 'phased' attribute only returns True or False, correct?
How can I obtain the phase set field (PS) from the VCF file?
Thanks!

**Comment by bioinformed** [comment]
@fiapinto: use `call['PS']` or `call.get('PS')`

**Comment by fiapinto** [comment]
VariantRecordSample.call['PS'] or  VariantRecordSample.phased.call['PS']?

**Comment by bioinformed** [comment]
No-- I'm assuming `call` is an instance of `VariantRecordSample`.  So it is `VariantRecordSample['PS']` or `VariantRecordSample.get('PS')` (since the class acts like a dictionary).

**Comment by fiapinto** [comment]
Got it! Thanks a lot!

**Comment by bjpop** [comment]
Sorry to comment on a closed item, but I believe this has not been solved:

http://pysam.readthedocs.io/en/latest/index.html

Searching for VariantRecordSample reveals no information. It looks like the docs are not up to date with the latest version of the program?

**Comment by ScottNortonPhD** [comment]
Bumping because this is still not resolved.

**Comment by PlatonB** [comment]
@mrvollger A way to preview VariantRecordSample structure by the first sample cell of any VCF:
```
from pysam import VariantFile
from pprint import pprint
variant_file_obj = VariantFile('/path/to/file.vcf.gz')
for rec in variant_file_obj.fetch():
        for sample_name in rec.samples:
                pprint(dict(rec.samples[sample_name]))
                break
        break
```
```
{'AD': (0, 80),
 'DP': 80,
 'GQ': 35,
 'GT': (1, 1),
 'PL': (37, 38, 0),
 'VAF': (1.0,)}
```

**Comment by regevs** [comment]
This is still not resolved - thanks.

**Comment by jmarshall** [comment]
It turns out that these need to be listed with `autoclass` in _doc/api.rst_ to be listed in the classes documentation. So we'll find the appropriate subset of and ordering for `Variant{Header,Record}*` classes to be added to the list.

**Comment by blex-max** [comment]
adding a bump on this one, had to use this thread/poking at the class for documentation

**Comment by benj5378** [comment]
I can use `VariantRecordSample["key"]` to get FORMAT attributes. Are there anyway to retrieve INFO attributes?
